File: assets_rebalance_backend/src/Boundaries/ChangeFinAssetBankAccountInput.cs
using System;
using assets_rebalance_backend.Domain;

namespace assets_rebalance_backend.Boundaries;

public class ChangeFinAssetBankAccountInput
{
    public Guid? Id { get; set; }
    public required Guid BankId { get; set; }
    public required string Name { get; set; }
    public IEnumerable<string> Tags { get; set; } = [];

    public FinAssetBankAccount Domain(FinAssetBank bank)
       => new()
       {
           Id = Id ?? Guid.NewGuid(),
           Bank = bank,
           Name = Name,
           Tags = Tags,
       };
}

File: assets_rebalance_frontend/src/lib/boundaries/change-fin-asset-bank-account.input.ts
import { FinAssetBank } from "@/lib/domain/fin-asset-bank";
import { FinAssetBankAccount } from "@/lib/domain/fin-asset-bank-account";

export class ChangeFinAssetBankAccountInput {
  id?: string
  bankId: string; // Required (UUID is represented as a string in TypeScript)
  name: string; // Required
  tags: string[] = []; // Default to an empty array

  constructor(bankId: string, name: string, tags: string[] = [], id?: string) {
    this.bankId = bankId;
    this.name = name;
    this.id = id
    this.tags = tags;
  }

  toDomain(bank: FinAssetBank): FinAssetBankAccount {
    return {
      id: this.id,
      bank: bank,
      name: this.name,
      tags: this.tags,
    };
  }
}
